cpp/src/parquet/statistics_test.cc

```diff
@@ -452,6 +452,13 @@ class TestStatistics : public PrimitiveTypedTest<TestType> {
     EXPECT_TRUE(expected_stats->HasMinMax());
     EXPECT_EQ(expected_stats->EncodeMin(), stats->EncodeMin());
     EXPECT_EQ(expected_stats->EncodeMax(), stats->EncodeMax());
+
+    std::shared_ptr<EncodedStatistics> enc_stats = column_chunk->encoded_statistics();
+    EXPECT_EQ(null_count, enc_stats->null_count);
+    EXPECT_TRUE(enc_stats->has_min);
+    EXPECT_TRUE(enc_stats->has_max);
+    EXPECT_EQ(expected_stats->EncodeMin(), enc_stats->min());
+    EXPECT_EQ(expected_stats->EncodeMax(), enc_stats->max());
```

Comment:
Could you perhaps also test the result of `encoded_statistics()` on a file that doesn't have statistics enabled?

Reply:
@pitrou did you mean like this?
https://github.com/apache/arrow/pull/46463/commits/13aa06eee80bf2e13ff7374fd6d970f50bbe2264

Reply:
Yes, that looks good to me.